packages/ODEs/src/app.ts

```diff
@@ -1,654 +1,870 @@
+      const step = ivp.arg.step.value;
+
+      if (start >= finish)
+        return;
+
+      if ((step <= 0) || (step > finish - start))
+        return;
+
+      const scriptText = getScriptLines(ivp).join('\n');
       const script = DG.Script.create(scriptText);
-      const params = getScriptParams(ivp);    
+      const params = getScriptParams(ivp);
       const call = script.prepare(params);
 
       await call.call();
-      
-      solutionTable = call.outputs[DF_NAME];
-      solverView.dataFrame = call.outputs[DF_NAME];
-      solverView.name = solutionTable.name;
-
-      if (!solutionViewer) {
-        solutionViewer = DG.Viewer.lineChart(solutionTable, lineChartOptions(solutionTable.columns.names()));
-        viewerDockNode = grok.shell.dockManager.dock(
-          solutionViewer, 
-          DG.DOCK_TYPE.TOP, 
-          solverView.dockManager.
-          findNode(solverView.grid.root
-        ));
+
+      if (this.solutionViewer)
+        this.solutionViewer.setOptions({segmentColumnName: (ivp.updates !== null) ? STAGE_COL_NAME: null});
+
+      this.solutionTable = call.outputs[DF_NAME];
+      this.solverView.dataFrame = call.outputs[DF_NAME];
+      this.solverView.name = this.solutionTable.name;
+
+      if (ivp.updates) {
+        this.solverView.grid.columns.setVisible([this.solutionTable.columns.names()[0]]);
+        this.solverView.grid.columns.setVisible(this.solutionTable.columns.names()
+          .filter((name) => name !== STAGE_COL_NAME));
       }
-      else {
-        solutionViewer.dataFrame = solutionTable;
 
-        if (toChangeSolutionViewerProps) {
-          solutionViewer.setOptions(lineChartOptions(solutionTable.columns.names()));
-          toChangeSolutionViewerProps = false;
+      if (!this.solutionViewer) {
+        this.solutionViewer = DG.Viewer.lineChart(this.solutionTable,
+          getLineChartOptions(this.solutionTable.columns.names()));
+        this.viewerDockNode = grok.shell.dockManager.dock(
+          this.solutionViewer,
+          DG.DOCK_TYPE.TOP,
+          this.solverView.dockManager.
+            findNode(this.solverView.grid.root),
+        );
+      } else {
+        this.solutionViewer.dataFrame = this.solutionTable;
+
+        if (this.toChangeSolutionViewerProps) {
+          this.solutionViewer.setOptions(getLineChartOptions(this.solutionTable.columns.names()));
+          this.toChangeSolutionViewerProps = false;
         }
       }
 
-    } catch (err) {
-        if (err instanceof Error) 
-          grok.shell.error(`${ERROR_MSG.SOLVING_FAILS}: ${err.message}`);
-        else
-          grok.shell.error(`${ERROR_MSG.SOLVING_FAILS}: ${ERROR_MSG.CORE_ISSUE}`);
-  }};
-
-  /** Run model application */
-  const runModelApp = async () => {
+      this.isSolvingSuccess = true;
+    } catch (error) {
+      this.clearSolution();
+
+      if (error instanceof Error)
+        grok.shell.error(error.message);
+      else
+        grok.shell.error(ERROR_MSG.SCRIPTING_ISSUE);
```

Comment:
you can make it in a ternary way
`grok.shell.error(error instanceof Error ? error.message : ERROR_MSG.SCRIPTING_ISSUE);`

Reply:
Done